haskell/private/pkg_id.bzl

```diff
@@ -0,0 +1,50 @@
+"""Package identifiers"""
+
+load("@bazel_skylib//:lib.bzl", "paths")
+
+def _zencode(s):
+  """Z-escape special characters to make a valid filename.
```

Comment:
It's not just for filenames; this also gives us a valid GHC package identifier.  (Otherwise it wouldn't be necessary to escape underlines.) . Clarify in the comment?

Reply:
True!